_posts/2017-08-02-July.md

```diff
@@ -0,0 +1,45 @@
+- Browser Client - The browser client connects to a remote Eve server to which it streams records like key presses; and receives records like ticks of a timer. These changes are rendered to the browser by watchers, which extend Eve's functionality into a foreign language. So far we have interfaces for writing watchers in Rust and Typescript.
+
+- Persistance - This is one of the features that has been missing for a while and frequently requested. Persistence is implemented by writing to an append-only binary file at a user specified location. Right now this is a flag passed to the Eve executable, but this could be controlled from within Eve itself. Transactions are written to this log as they are executed in the runtime. On load, the log of transactions is replayed, which restores the runtime to the state it was at when it was terminated. There are some obvious limitations to this when we start talking about Eve in a distributed context, but this at least gets us a starting place.
+
+What about crashes?
+
+- Library - The entire standard library has to be re-implemented in Rust. This is largely a matter of just the work of wrapping native Rust functions, but we want to hold off on that until we implement a custom number system we have planned (more on that in a future post). We've added old watchers like file, console, and compiler; as well as new watchers like http and json. These newest additions allow you to parse JSON files into records, or send HTTP requests to API endpoints and parse the response.
+
+![Error](/images/v0.4-error.png)
+
+- Errors - We are reporting errors in the console, including line numbers and a helpful hint. We will also make these errors available through a request to the server, allowing 3rd party developer/editor extensions to integrate error reporting.
+
+- More testing - Rust has really nice testing facilities that make writing and running unit tests very easy. We've written a bunch of those that broadly cover 
+
+- Packaging and Distribution - To reach a wider audience, packaging and distribution will be a primary focus for this release. Packaging the Eve server is mostly handled for us by Rust, as it emits executables for all major platforms. The server can even host a native Eve editor. We've also experimented with embedding the Eve server in an Electron app. This solution allows us to present Eve as more of a native application, separate from the browser (even though technically...).
+
+## v0.4 timeline
+
+When it became clear that despite our move to an incremental evaluation strategy, v0.3 wouldn't be fast enough to run the database explorer, we started work on the Rust runtime in hopes that a lower-level host language would offer greater speed improvements. Now it looks like the Rust runtime is here to stay, and we will release it as v0.4 by the end of the Summer. Right now the core runtime is more-or-less complete, and we are working hard at making the product easy to use. 
```

Comment:
We're basically doing final bug fixes and tooling work to make it ready for general use.